jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -288,31 +288,28 @@ protected boolean responseHeaders(HttpExchange exchange)
         HttpResponse response = exchange.getResponse();
         if (LOG.isDebugEnabled())
             LOG.debug("Response headers {}{}{}", response, System.lineSeparator(), response.getHeaders().toString().trim());
+
+        // Content-Encoding may have multiple values in the order
+        // they are applied, but we only support the last one.
+        List<String> contentEncodings = response.getHeaders().getCSV(HttpHeader.CONTENT_ENCODING.asString(), false);
+        if (contentEncodings != null && !contentEncodings.isEmpty())
+        {
+            // Pick the last content encoding from the server.
+            String contentEncoding = contentEncodings.get(contentEncodings.size() - 1);
+            decoder = getHttpDestination().getHttpClient().getContentDecoderFactories().stream()
+                .filter(f -> f.getEncoding().equalsIgnoreCase(contentEncoding))
+                .findFirst()
```

Comment:
should this be the last not the first?

Reply:
@gregw I think you are looking at outdated code.

Reply:
oops